### PR TITLE
HelloWorld-NuGet

### PR DESCRIPTION
**Pull request by jthelin**
HelloWorld-NuGet
- Convert the HelloWorld sample to use the Orleans NuGet packages.
- Enable NuGet package restore for this .sln.
- Run DevTest silo from the client exe prog dir, and don't write extra
stuff from this .sln into the LocalSilo directory.

### CODE REVIEW

**Comment by jthelin-microsoft** [comment]
See discussion in issue #125

File: Samples/HelloWorld/HelloWorld/HelloWorld.csproj [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
     <FileAlignment>512</FileAlignment>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
+    <RestorePackages>true</RestorePackages>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>$(OrleansSDK)\LocalSilo\</OutputPath>
+    <OutputPath>bin\$(Configuration)\</OutputPath>
```
Comment (gabikliot) [inline Samples/HelloWorld/HelloWorld/HelloWorld.csproj]: why not $(OrleansSDK)\LocalSilo?

Comment (jthelin-microsoft) [inline Samples/HelloWorld/HelloWorld/HelloWorld.csproj]: If you are just using the packages downloaded from NuGet.org, then OrleansSDK may not be defined. The DevTest Silo can run in-place from anywhere, and does not need to be conflated with LocalSilo [if the full SDK installed]  With this change, the "client" program now starts silo in background from its own bin directory.

Comment (gabikliot) [inline Samples/HelloWorld/HelloWorld/HelloWorld.csproj]: 1) I think this is a big enough change in the way we run Samples that we want to discuss it before making any changes to it.
2) This change by itself might be a good idea, or not. I am concerned that now we can't run samples against the latest non official build, but only against the officially published nugets. 
3) We are still thinking about installing SDK as the main way to use Orleans, and nugets are there just to support those who want to consumer from nugets.
4) Not sure the whole change is worth our time and efforts, given so many other higher priority tasks.

**Comment by MovGP0** [comment]
I am unhappy with both variants. 

* I want to remove the dependency that the OrleansSDK environment variable is defined.
* I want to deploy somewhere within the OrleansSDK folder when it exists.
* I don't want the deployment into the OrleansSDK folder directly, so that the binaries from Orleans and the examples are separated.

I therefore suggest to do something like the following: 

    <Warning 
        Text="The OrleansSDK environment variable is not set. For details see http://orleans.codeplex.com/wikipage?title=Orleans%20Setup%20for%20Developers" 
        Condition="!Exists('$(OrleansSDK)')" /> 
    <OrleansSDK Condition="!Exists('$(OrleansSDK)')">bin\$(Configuration)\<OrleansSDK>
    
    <OutputPath>$(OrleansSDK)\LocalSilo\HelloWorld\</OutputPath>
    
    <MakeDir Directories="$(OutputPath)"/> Condition="!Exists('$(OutputPath)')" />

**Comment by jthelin-microsoft** [comment]
When i have some spare time, I will try to work out an updated change set that preserves some of the optionality behavious that @MovGP0 describes.

**Comment by sergeybykov** [comment]
Discussed this with @jthelin-microsoft. We agreed to close this PR for now to prevent confusion about how samples are supposed to run. Will likely resurrect later in a modified form.

**Comment by jthelin** [comment]
I have posted some notes in the project wiki about how to convert legacy Orleans .csproj to use the new Orleans NuGet packages. 

We will also update the Orleans vsix templates to use NuGet packages going forward.

https://github.com/dotnet/orleans/wiki/Convert-Orleans-v0.9-csproj-to-Use-v1.0-NuGet

**Comment by ksuszka** [comment]
I just tried to follow those instructions but it didn't really work for me. I'm not able to compile converted interfaces project. It seems that NuGet package is broken - Microsoft.Orleans.Templates.Interfaces.targets file tries to find ClientGenerator.exe tool in "..\tools" directory, which is not present. I think it should point to Microsoft.Orleans.ClientGenerator.1.0.3.

**Pull request closed without merging**